cmd/cuebe/cmd/export.go

```diff
@@ -48,73 +55,93 @@ cuebe export -i main.enc.yaml
 	}
 
 	f := cmd.Flags()
-	f.StringSliceP("inject", "i", []string{}, "Raw YAML files to inject. Can be encrypted with sops.")
-	f.StringP("expression", "e", "", "Expressions to extract manifests from. Extract all manifests by default.")
-	f.StringArrayP("tag", "t", []string{}, "Inject boolean or key=value tag.")
-	f.StringP("path", "p", "", "Path to load CUE from. Default to current directory")
+	flag.AddBuild(f)
+	f.StringP("output", "o", "", "Directory to write manifests to. If empty it will print into stdout")
 	return cmd
 }
 
-func exportCmd(cmd *coral.Command, args []string) {
+func exportCmd(cmd *cobra.Command, args []string) {
 	opts, err := exportParse(cmd, args)
-	coral.CheckErr(err)
-	coral.CheckErr(exportRun(cmd, opts))
+	cobra.CheckErr(err)
+	cobra.CheckErr(exportRun(cmd, opts))
 }
 
-func exportParse(cmd *coral.Command, args []string) (*exportOpts, error) {
-	opts := &exportOpts{}
+func exportParse(cmd *cobra.Command, args []string) (*exportOpts, error) {
+	opts := new(exportOpts)
 
-	// InjectFiles
-	i, err := cmd.Flags().GetStringSlice("inject")
+	bopts, err := flag.GetBuild(cmd.Flags())
 	if err != nil {
-		return nil, fmt.Errorf("Failed parsing args: %w", err)
+		return opts, fmt.Errorf("could not get build options: %w", err)
 	}
-	opts.InjectFiles = i
+	opts.BuildOpt = *bopts
 
-	// Expression
-	e, err := cmd.Flags().GetString("expression")
+	// output
+	output, err := cmd.Flags().GetString("output")
 	if err != nil {
-		return nil, fmt.Errorf("Failed parsing args: %w", err)
+		return nil, fmt.Errorf("could not get output flag: %w", err)
 	}
-	opts.Expression = e
+	opts.OutputDir = output
+
+	// context
+	opts.BuildContext = bctxt.New()
+	for _, arg := range args {
+		// TODO move that to a function in the context package
+		if !path.IsAbs(arg) {
+			cwd, err := os.Getwd()
+			if err != nil {
+				return opts, fmt.Errorf("could not get working directory: %w", err)
+			}
+			arg = path.Join(cwd, arg)
+		}
+		if err := opts.BuildContext.Add(afero.NewBasePathFs(afero.NewOsFs(), arg)); err != nil {
```

Comment:
Just so I understand correctly what's happening, `.Add()` will copy each context (passed as command args) to the `BuildContext` which is a `fs` or type `MemMapFs`. So Afero returns you an "InMemory" FS when instantiated with `bctxt.New()`, then .Add will copy each detected context in that memory space for future operation?